src/LinearSolverTypes.f90

```diff
@@ -236,10 +264,12 @@ SUBROUTINE linearsolver_sub_absintfc(solver)
     PROCEDURE,PASS :: solve => solve_LinearSolverType_Iterative
     !> @copybrief LinearSolverTypes::getResidual_LinearSolverType_Iterative
     !> @copydetails LinearSolverTypes::getResidual_LinearSolverType_Iterative
-    PROCEDURE,PASS :: getResidual_LinearSolverType_Iterative
-    PROCEDURE,PASS :: getIterResidual_LinearSolverType_Iterative
-    GENERIC :: getResidual => getResidual_LinearSolverType_Iterative, &
```

Comment:
why did this generic interface go away?

Reply:
It didn't go away, exactly. It was generalized to the LinearSolverType_Base, which specifies an interface "getResidual" to be implemented by the child classes (LinearSolverType_Direct and iterative).
A residual is still computable for direct solvers, and this generalization cleans up the CMFD code in MPACT significantly, especially in the getConvNorm subroutines.

Reply:
yeah, ok....moving to the base type is fair

Reply:
so which interfaces(s) had the issue with switching the order of the arguments?

Reply:
It is this one, the getResidual.

I'm working on a patch for this (it was delayed by the bwr thermal expansion branch). The goal was to make these subroutines have all optional arguments so that the calls in MPACT could be kept, and things would not break in the transition.
Unfortunately that won't work out, since making both of the subroutines in the interface have optional arguments makes this ambiguous.
So the solution is to replace the interface with a single function with 3 optional arguments (iteration count, residual norm, and residual vector). These can be ordered to conform to the function calls to MPACT (I believe) and the Futility patch conventions can be switched over to these. This should not take much longer, hopefully there will be some changes pushed up today/tomorrow.

Reply:
Another option would be to have a couple versions of the routine behind a GENERIC interface.  One version of the generic would match the old version that MPACT calls.  Only one version would have to actually do something: the rest could simply wrap the main one, passing in temporary variables just to match the interface.  Sometimes the GENERIC interface is cleaner than having a bunch of optional arguments.

Reply:
That might work too, but it's been fixed now.
However, another incompatibility has popped up with the changes to VectorResemble. The old code passed in a pointer, but it's been modified now to accept an allocatable.
Unfortunately, in Fortran 95 a distinction can't be made in the function signature between an allocatable and a pointer (in '03 or '08 you can) so this is a bit trickier to deal with but I think it can be done with a (relatively) simple patch in MPACT thanks to the native distributed vector type which is already in.

Reply:
I'm really nervous about how many interface changes are going on in this pull request.  Is this allocatable/pointer change really necessary?  It doesn't seem like it to me from reading the code.

Reply:
I can't speak to how necessary it is for this MR, however, the addition of resemble being able to handle allocatable vectors would be helpful.

Reply:
There should only be 2; getResidual and this one.
I adjusted this one for 2 reasons: It is called only once in all of the MPACT code (CMFD_Base) and it allows me to use the VectorResemble routine to allocate non-pointer vector types. Which is important, as making everything in the linear solver a pointer or target is just not practical.

Reply:
Maybe there needs to just be `VectorResemble` and `VectorResemble_Alloc`.  Having both pointer and allocatable versions could be useful, especially since you can't apply the TARGET attribute to an allocatable array.

Reply:
In all fairness, it probably should have started as an allocatable anyways... some of the early mpact/futility code is pretty pointer-happy.  I'm just not a fan of simultaneous merges in multiple repos

Reply:
The different name should work. I'm not sure why I didn't think of that; it's a lot less work than the fix I'm trying to make.

Reply:
An update on this; the newest build causes no incompatibilities, and all unit tests in MPACT pass. I'm doing some more in-depth tests (Heavies, unused variables, serial versions) but it's looking good.